src/test/java/de/eldoria/semvertools/util/IntegersTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2021 SemVerTools team and contributors
+ */
+
+package de.eldoria.semvertools.util;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class IntegersTest {
+
+  @Test
+  void testParseEmpty() {
+    assertEquals(Optional.empty(), Integers.parseNonNegativeInt(""));
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"00", "01", "09", "000", "00001"})
+  void testParseLeadingZero(String value) {
+    assertEquals(Optional.empty(), Integers.parseNonNegativeInt(value));
```

Comment:
```suggestion
    assertTrue(Integers.parseNonNegativeInt(value).isEmpty());
```

Reply:
I prefer `assertEquals` over `assertTrue` here because when failing, it prints the two values. That's more explanative than just having a `expected true but got false`.